rust/otap-dataflow/crates/controller/src/lib.rs

```diff
@@ -321,11 +329,25 @@ impl<PData: 'static + Clone + Send + Sync + std::fmt::Debug> Controller<PData> {
         pipeline_ctrl_msg_tx: PipelineCtrlMsgSender<PData>,
         pipeline_ctrl_msg_rx: PipelineCtrlMsgReceiver<PData>,
     ) -> Result<Vec<()>, Error> {
+        // Create a tracing span for this pipeline thread
```

Comment:
@lquerel @jmacd Showing how we can contextual enrichment by specifying these info once, and it'll auto-inherit for all subsequent logs.

If needed, these can be converted to proper OTel spans as well. (using a 3rd party crate named `tracing-opentelemetry`)